didman/api/v1/api.go

```diff
@@ -94,31 +94,50 @@ func (w *Wrapper) Routes(router core.EchoRouter) {
 	}))
 }
 
-// AddEndpoint handles calls to add a service. It only checks params and sets the correct return status code.
-// didman.AddEndpoint does the heavy lifting.
-func (w *Wrapper) AddEndpoint(ctx context.Context, request AddEndpointRequestObject) (AddEndpointResponseObject, error) {
-	id, err := did.ParseDID(request.Did)
+func (w *Wrapper) addOrUpdateEndpoint(
```

Comment:
Function params gives extra mental complexity to prevent just a few duplicate lines.

Reply:
First complain about duplicate logic, then complain about generic functions to prevent code duplication? :-p